client/src/modules/InterviewQuestions/components/Modals/CategoryModalForm/index.tsx

```diff
@@ -0,0 +1,84 @@
+import { Form, Input, message, Select } from 'antd';
+import { Option } from 'antd/lib/mentions';
+import Modal from 'antd/lib/modal/Modal';
+import { useEffect } from 'react';
+import { InterviewQuestionCategoryService } from 'services/interviewQuestion';
+import { InterviewQuestion, InterviewQuestionCategory } from 'services/models';
+
+type Props = {
+  questions: InterviewQuestion[];
+  category: InterviewQuestionCategory | null;
+  isVisible: boolean;
+  onCancel: () => void;
+  loadData: () => Promise<void>;
+};
+
+interface IValues {
+  name: string;
+  questions: string[];
+}
+
+export function CategoryModalForm(props: Props) {
+  const { category, isVisible, onCancel, questions, loadData } = props;
+  const [form] = Form.useForm();
+  const interviewQuestionCategoryService = new InterviewQuestionCategoryService();
+
+  useEffect(() => {
+    return () => {
+      form.resetFields();
+    };
+  }, [isVisible]);
+
+  const initialValues = {
+    name: category?.name,
+    questions: category?.questions.map(el => el.title),
+  };
+
+  const createRecord = (values: IValues) => {
+    return {
+      name: values.name,
+      questions: values.questions?.reduce((acc, el) => {
+        const question = questions.find(question => question.title === el);
+        if (question) acc.push(question);
+        return acc;
+      }, [] as InterviewQuestion[]),
+    };
+  };
+
+  const submitCategory = async () => {
+    try {
+      const values = await form.validateFields().catch(() => null);
+      const record = createRecord(values);
```

Comment:
I don't see that createRecord apply null as an argument and do something if null has been passed. And anyway error will get caught at 58 line. So just delete this `.catch(() => null)`